lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3328,15 +3329,101 @@ LogicalResult ConvertAtenOp<AtenIndexTensorOp>::matchAndRewrite(
   if (!getListConstructElements(tensorList, tensorsTorchType))
     return op.emitError(
         "unimplemented: the tensor list is not from list construct");
-  auto tensors = getTypeConvertedValues(rewriter, op->getLoc(),
-                                        getTypeConverter(), tensorsTorchType);
+  auto indexTensors = getTypeConvertedValues(
+      rewriter, op->getLoc(), getTypeConverter(), tensorsTorchType);
 
-  // TODO add support for multiple index
-  if ( tensors.size() > 1){
-    return op.emitError(
-        "unimplemented: the index tensor list from list construct > 1");
+  auto outType = getTypeConverter()->convertType(op.getType());
+
+  // Support for multiple indexes
+  if (indexTensors.size() > 1) {
+    // t[i, i]
+    // = torch.ops.aten.index(t,(i,i))
+    // = tensor([[ t[1,1], t[2,2], t[3,3]],
+    //           [ t[3,3], t[2,2],  t[1,1]]])
+    // = tensor([[ 7, 13, 19], [19, 13,  7]])
+    // = tf.gather_nd(t,tf.ii_expand)
+    // ii_expand
+    // = tf.concat((i_expand,i_expand), dim=2)
+    // = tf.constant([[[1,1],[2,2],[3,3]],
+    //                [[3,3],[2,2],[1,1]]]) # 2*3*2
+    SmallVector<Value> indicesTfConcatTensors;
+    SmallVector<int64_t> indexesRank;
+    SmallVector<SmallVector<int64_t>> indexesShape;
+
+    // concat index tensor into to indices tensor for concat
+    for (auto index : indexTensors) {
+      // TODO add support for none index input like torch.ops.aten.index(x,
+      // (None, index1, index2, None))
+      if (!index.getImpl())
```

Comment:
drive by comment: I don't think this will actually check if the tensor is `None`. The way that this has been done in torch-mlir is by checking that the type is `NoneType`. For example:

https://github.com/llvm/torch-mlir/blob/320e67ff3477dda2c1a03fe2d4d230774586eedc/lib/Conversion/TorchToLinalg/DataMovement.cpp#L40

Reply:
The getType will call impl->getType(); But in my case, the impl is null, so it will return segmentation fault.

Reply:
done